manager-rs/dauth-service/src/tasks/task_manager.rs

```diff
@@ -33,8 +33,30 @@ async fn run(context: Arc<DauthContext>) -> Result<(), DauthError> {
         if let Err(e) = tasks::register::run_task(context.clone()).await {
             tracing::warn!("Failed to run register task: {}", e);
         } else {
-            if let Err(e) = tasks::update_users::run_task(context.clone()).await {
-                tracing::warn!("Failed to run update user task: {}", e)
+            let mut tasks = Vec::new();
+
+            tasks.push(tokio::spawn(tasks::update_users::run_task(context.clone())));
+            tasks.push(tokio::spawn(tasks::replace_key_shares::run_task(
+                context.clone(),
+            )));
+            tasks.push(tokio::spawn(tasks::report_auth_vectors::run_task(
+                context.clone(),
+            )));
+            tasks.push(tokio::spawn(tasks::report_key_shares::run_task(
+                context.clone(),
+            )));
+
+            for task in tasks {
+                match task.await {
```

Comment:
The task thread is separate from server/handler thread. It spawns tasks to run separately and waits for them all to finish before looping and checking again. What do you mean by missing ticks?

Reply:
Ah, I see. I had thought the task thread would be polling periodically at a fixed interval, but the current design has it polling at a fixed spacing. (I.e. it polls every max(time_to_complete_task, 10s) vs. every 10s regardless of time to complete). I think this is okay actually for what we're trying to accomplish since we don't need these background tasks to be very responsive, and it allows the design to not have to worry about if a task is currently in process or not.